api/tests/Feature/TeamsTest.php

```diff
@@ -255,4 +279,117 @@ public function testTeamUpdateMutation(): void
+  {
+    // Detach departments before deleting team
+    $this->team3->departments()->detach();
+
+    // Assert team update successful across all input fields
+    $this->actingAs($this->admin, "api")->graphQL(
+      /** @lang GraphQL */
+      '
+      mutation deleteTeam($id: UUID!) {
+          deleteTeam(id: $id) {
+            id
+          }
+      }
+      ',
+      [ 'id' => $this->toBeDeletedUUID, ]
+    )->assertJson([
+      'data' => [
+        'deleteTeam' => [
+          'id' => $this->toBeDeletedUUID,
+        ],
+      ]
+    ]);
+  }
+
+  public function testViewAnyTeamsPolicy(): void
+  {
+    // Create new users and assign them to team
+    $poolOperator1 = User::factory()->create([
+      'email' => 'poolOperator1@test.com',
+      'sub' => 'poolOperator1@test.com',
+    ]);
+
+    $poolOperator1->syncRoles([
+      "guest",
+      "base_user",
+      "applicant",
+      "request_responder",
+      "platform_admin"
+    ]);
+
+    $poolOperator2 = User::factory()->create([
+      'email' => 'poolOperator2@test.com',
+      'sub' => 'poolOperator2@test.com',
+    ]);
+
+    $poolOperator2->syncRoles([
+      "guest",
+      "base_user",
+      "applicant",
+      "request_responder",
+      "platform_admin"
+    ]);
+
+    $poolOperator3 = User::factory()->create([
+      'email' => 'poolOperator3@test.com',
+      'sub' => 'poolOperator3@test.com',
+    ]);
+
+    $poolOperator3->syncRoles([
+      "guest",
+      "base_user",
+      "applicant",
+      "request_responder",
+      "platform_admin"
+    ]);
+
+    $poolOperatorRole = Role::where('name', 'pool_operator')->sole();
+    $poolOperator1->attachRole(
+        $poolOperatorRole,
+        $this->team1
+    );
+    $poolOperator2->attachRole(
+        $poolOperatorRole,
+        $this->team1
+    );
+    $poolOperator3->attachRole(
+        $poolOperatorRole,
+        $this->team1
+    );
+
+    // Assert all teams query contains expected results
+    $query = $this->actingAs($this->admin, "api")
+      ->graphQL(/** @lang Graphql */ '
+          query team($id: UUID!) {
+            team(id: $id) {
+                id
+                roleAssignments {
+                  id
+                  user {
+                    id
+                  }
+                }
+            }
+          }
+    ', [ 'id' => $this->team1->id ]);
+
+    $data = $query->original['data'];
+
+    // assert the teams returned is an array of exactly three items
+    $teamMembersCount = count($data['team']['roleAssignments']);
+    assertEquals($teamMembersCount, 3);
+
+    // assert every team member is present in the response
+    $query->assertJsonFragment(['id' => $poolOperator1->id]);
+    $query->assertJsonFragment(['id' => $poolOperator2->id]);
+    $query->assertJsonFragment(['id' => $poolOperator3->id]);
+  }
+
+
+  // TODO: Create another test policy for view-team-teamMembers where we fetch roleAssignments (users) from a specific team (?).
```

Comment:
Yeah, this would be good. You can ask for some pairing help if you want.

Reply:
... I think I got it 😅. Take a look at the Policy and the test and let me know! The test only covers the policy and not any mutation since it hasn't been used in the api yet.

Reply:
I'm not sure when it got merged into main, but it actually is used in the API! Take a look at Team.roleAssignments:
![image](https://user-images.githubusercontent.com/4313717/226481245-d2743bec-bec2-4610-b30e-0d6f807fca7b.png)

A good query to test would be 
```
teams {
  roleAssignments {
    role { name }
    user { 
      id
      email
    }
  }
}
```

Reply:
Oh, nevermind, I just realized you're already doing that query in `testViewAnyTeamMembers`

Reply:
Nevermind x2, I just realized that Team.roleAssignments checks view**Any**TeamMembers, not viewTeamMembers. 🤔

Reply:
Okay, I think the solution is to change the Team.roleAssignments policy from viewAnyTeamMembersto the more specific viewTeamMembers, and then rename testViewAnyTeamMembers to testViewTeamMembers, while also checking that the query works when acting as a PoolOperator in the same pool (and not as a PoolOperator in a different pool).

Reply:
> Nevermind x2, I just realized that Team.roleAssignments checks view**Any**TeamMembers, not viewTeamMembers. 🤔

😆 ya that's why I left that note about not testing any query. Okay I'll get that done thanks Tristan.